### PR TITLE
[TEAM2-500] Filter User Assets By Socket Support In Currency Select Modal

### PR DESCRIPTION
**Pull request by derHowie**
## What changed (plus any additional context for devs)
When a destination network can be determined based on output token selection

## Screen recordings / screenshots
https://recordit.co/avBPAe34ie

## What to test
The team 2 wallet contains a currency (Pikatic) that does not currently possess on supported cross chain routes. This is the most accessible test case. You can also use the curl example below to verify which tokens are supported for a given chain combination, swap for something unsupported, and attempt the flow I illustrated above with Pikatic.

```
curl --location --request GET 'https://api.socket.tech/v2/token-lists/from-token-list?fromChainId=10&toChainId=1' --header 'API-KEY: 645b2c8c-5825-4930-baf3-d9b997fcd88c'
```
This request returns tokens that can be swapped from optimism (10) to mainnet (1).

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/hooks/useSwapCurrencyList.ts
Comment (derHowie) [inline src/hooks/useSwapCurrencyList.ts:370]: This is unrelated to the feature. L2 search results on this feature branch were retaining mainnet's results in categories outside of verified assets. This fixes that.

File: src/handlers/tokenSearch.ts
Comment (estebanmino) [inline src/handlers/tokenSearch.ts:63]: any reason to change the formatting of these names?

https://github.com/rainbow-me/rainbow/pull/4192/files#diff-fc39c286bdc8768f081731a684b5ed51a1c3c6c31125422376fbf98c12090fabR22

`fromChainId` and `toChainId` lgtm

File: src/handlers/tokenSearch.ts
Comment (estebanmino) [inline src/handlers/tokenSearch.ts:71]: ah yes like here, re: naming

File: src/hooks/useSwapCurrencyList.ts
Comment (estebanmino) [reply src/hooks/useSwapCurrencyList.ts:370]: 🚀

File: src/hooks/useSwapDerivedOutputs.ts
Comment (estebanmino) [inline src/hooks/useSwapDerivedOutputs.ts:146]: this was the flipping fix on my branch right?

File: src/hooks/useSwappableUserAssets.ts
Comment (estebanmino) [inline src/hooks/useSwappableUserAssets.ts:98]: ok i think this is related https://github.com/rainbow-me/rainbow/pull/4192/files#r982956366, so we're safe doing this?

File: src/hooks/useSwapDerivedOutputs.ts
Comment (estebanmino) [reply src/hooks/useSwapDerivedOutputs.ts:146]: ah yea i see the cherry-pick

File: src/hooks/useSwappableUserAssets.ts
Comment (estebanmino) [inline src/hooks/useSwappableUserAssets.ts:101]: aren't ETH_ADDRESS and ETH_ADDRESS_AGGREGATORS the same? i'm confused https://github.com/rainbow-me/rainbow/pull/4221/files#diff-6a86a7622d34d66716c4521a890f7a81aaca4a1904c2869ebebeab4600b54cb5R8

File: src/hooks/useSwappableUserAssets.ts
Comment (estebanmino) [inline src/hooks/useSwappableUserAssets.ts:39]: i think `addresses` is not used

File: src/hooks/useSwappableUserAssets.ts
Comment (estebanmino) [reply src/hooks/useSwappableUserAssets.ts:101]: ah nvm i'm blind, had to log it to figure it out

File: src/hooks/useSwappableUserAssets.ts
Comment (estebanmino) [reply src/hooks/useSwappableUserAssets.ts:98]: went in details logging stuff here and it was looking good 👍

File: src/screens/CurrencySelectModal.tsx
Comment (estebanmino) [inline src/screens/CurrencySelectModal.tsx:160]: wen ship

File: src/handlers/tokenSearch.ts
Comment (derHowie) [reply src/handlers/tokenSearch.ts:63]: Because `toChainId` and `fromChainId` are flipped depending on whether you are filtering the user's wallet assets vs requesting filtered tokens via search requests, I found it sort of unintuitive and confusing, so I named these something different to make it clear.

I'm fine with switching these to `fromChainId` and `toChainId`, though.

Comment (estebanmino) [reply src/handlers/tokenSearch.ts:63]: perfect, sounds good then!

File: src/hooks/useSwappableUserAssets.ts
Comment (derHowie) [reply src/hooks/useSwappableUserAssets.ts:39]: ahh, I was generating an address list in two place. Probably a remnant of my original implementation prior to refactor. This value is now being used.

File: src/screens/CurrencySelectModal.tsx
Comment (derHowie) [reply src/screens/CurrencySelectModal.tsx:160]: sooon 👀

File: src/handlers/tokenSearch.ts
Comment (derHowie) [reply src/handlers/tokenSearch.ts]: okay this has been changed to `toChainId` and `fromChainId`

**Review by estebanmino — APPROVED** [review]
the 🚢 oor

**Comment by estebanmino** [comment]
oh, could you hold off merging until the base branch is merged into develop? so we have 2 clean PRs merged

File: src/handlers/tokenSearch.ts
Comment (sammdec) [inline src/handlers/tokenSearch.ts:73]: Should we be returning a empty or null value here

**Review by sammdec — APPROVED** [review]
One comment about returning undefined in the walletFilter function but other than that looks good

File: src/handlers/tokenSearch.ts
Comment (derHowie) [reply src/handlers/tokenSearch.ts:73]: Thanks for highlighting this. It caused me to realize that we were not handling the failure case well at all (if this call blows up for some reason all tokens would be unavailable for swap).

I decided to let the error bubble up and handle this failure case at the call site, by allowing all tokens to be swapped rather than none. This seems like the lesser of two evils.

Comment (sammdec) [reply src/handlers/tokenSearch.ts:73]: Yeah sounds good to me 👍

**Review by sammdec — APPROVED** [review]
New changes lgtm

**Review by estebanmino — APPROVED** [review]

**Pull request merged**
